### PR TITLE
feat(sage-monorepo): update PR description skill (SMR-755)

### PR DESCRIPTION
**Pull request by sagely1**
## Description

Update skill to align with the new template which includes a Testing section

## Related Issue

[SMR-755](https://sagebionetworks.jira.com/browse/SMR-755)

## Changelog

  Structural changes:
  - Replaced "Preview" section with "Testing" (human-verifiable steps, no CI commands)
  - Added "Screenshots" sub-section under Testing (before/after table, only for visual changes)
  - Single output file at tmp/pr-description/<TICKET-ID>.md with title as H1

  Jira integration:
  - Discovers ticket ID from branch name → commit messages → user input
  - Supports all repo prefixes (AG, MG, QTL, SMR)
  - Expects Jira MCP; hard-stops on auth errors; degrades gracefully if no MCP configured
  - Fetches sub-tasks, linked issues, and acceptance criteria (feeds into Testing)

  Title generation:
  - Follows lint-pr.yml rules with concrete examples from the repo
  - Lowercase subject, ticket IDs appended in parentheses

  Removed:
  - Technology-specific checklists (redundant with analysis step)
  - github-pull-request_activePullRequest MCP reference
  - "When to use this skill" section (moved to frontmatter description)

## Testing

- After the Jira MCP is added in [SMR-722](https://sagebionetworks.jira.com/browse/SMR-722), we should see Jira info being used

[SMR-755]: https://sagebionetworks.jira.com/browse/SMR-755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SMR-722]: https://sagebionetworks.jira.com/browse/SMR-722?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .claude/skills/pr-description/SKILL.md [hunk 1]
```diff
@@ -1,186 +1,216 @@
 ---
 name: pr-description
-description: Generates comprehensive pull request descriptions following repository template standards. Use when asked to create, update, or generate PR descriptions, or when working with active pull requests that need documentation.
+description: >
+  Generates a PR title and description body following this repository's conventions.
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md]: We still would like it to follow our template pattern if possible.
```suggestion
  Generates a PR title and description body following repository template standards.
```

File: .claude/skills/pr-description/SKILL.md [hunk 2]
```diff
@@ -1,186 +1,216 @@
...
+verify to gain confidence this PR is correct.
+
+- Use a **task-list format** (`- [ ]`) so reviewers can check items off
+- Pull from Jira acceptance criteria when available — translate them into concrete reviewer steps
+- Focus on functional verification: navigation flows, data correctness, edge cases
+- Include API verification steps if relevant (e.g., "call GET /api/x and confirm the response includes field Y")
+- **Do NOT suggest running unit tests, e2e tests, or lint commands** — CI covers those
+- Aim for **3–6 actionable steps**
+
+### Screenshots
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md]: how about changing this section to "Preview" as we use recording as well.

Could you add this section to the template as well

File: .claude/skills/pr-description/SKILL.md [hunk 3]
```diff
@@ -1,186 +1,216 @@
...
+### Description
+
+- Lead with the business or technical objective — why was this work done?
+- Mention the Jira ticket context if it adds clarity
+- Keep it to 2-4 sentences
+
+### Related Issue
+
+- Always link the Jira ticket with a full URL: `[MG-893](https://sagebionetworks.jira.com/browse/MG-893)`
+- List sub-tasks and related issues underneath, indented, with their summaries
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md]: Do we want to include this here? I think the skill only needs to retrieve the relevant information from the tickets during creation, rather than listing all sub-tasks and related issues in this section.

**Review by rrchai — CHANGES_REQUESTED** [review]
Is the current PR description generated by this skill? It looks like it categorizes changelog into groups. Do we want to include that categorization behavior?

File: .claude/skills/pr-description/SKILL.md [hunk 4]
```diff
@@ -1,186 +1,216 @@
...
 
-## Related Issue
+This skill expects a Jira MCP server (e.g., `mcp-jira-cloud`) to be configured. Use the
+MCP tools to fetch the ticket. Pull:
 
-[If applicable, link to Jira ticket or GitHub issue]
-Fixes [ISSUE-NUMBER](link-to-issue)
+- Summary, description, and acceptance criteria
+- All **sub-tasks** (fetch them individually to get their full details)
+- All **linked/related issues**
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md:56]: this section probably could be removed after #4061 is merged, or specify to follow the rules in`.claude/rules/atlassian-mcp.md` (in case the rule is not automatically loaded for some reason)

File: .claude/skills/pr-description/SKILL.md [hunk 5]
```diff
@@ -1,186 +1,216 @@
...
-- Build system changes
+## Related Issue
 
-### For Python Changes
+[Link to the Jira ticket. Include sub-tasks and related issues if they exist.]
 
-Include when relevant:
+- [TICKET-ID](https://sagebionetworks.jira.com/browse/TICKET-ID)
+  - Sub-task: [TICKET-ID](url) — summary
+  - Related: [TICKET-ID](url) — summary
```
Comment (rrchai) [inline .claude/skills/pr-description/SKILL.md:134]: would your like to have all related tickets linked here or just main ticket? - I'm open to both.

File: .claude/skills/pr-description/SKILL.md (hunk 3 above)
Comment (sagely1) [reply .claude/skills/pr-description/SKILL.md]: Good idea.  Removing

File: .claude/skills/pr-description/SKILL.md (hunk 5 above)
Comment (sagely1) [reply .claude/skills/pr-description/SKILL.md:134]: Let's leave as is for now.  I don't have a strong opinion right now.

File: .claude/skills/pr-description/SKILL.md [hunk 6]
```diff
@@ -1,186 +1,215 @@
...
+- Keep items coarse-grained — group related file-level changes into one bullet
+- Order from most to least important
+- Aim for **3–7 bullets**; this is not an exhaustive file list
+
+### Testing
+
+This section is for **human reviewers**, not CI. Think about what a reviewer should manually
+verify to gain confidence this PR is correct.
+
+- Use a **task-list format** (`- [ ]`) so reviewers can check items off
```
Comment (hallieswan) [inline .claude/skills/pr-description/SKILL.md]: I would probably prefer bullets over the task-list format, since we generally have >1 reviewer per PR, so the ability to check items off isn't very useful. Up to you though! We could also consider splitting the testing section into different sub-steps for larger features (e.g. how to navigate through the UI to test particular functionality), but probably worth waiting to do that work until we run into a use case that we can use to fine tune the skill/template accordingly.

File: .claude/skills/pr-description/SKILL.md [hunk 7]
```diff
@@ -1,186 +1,215 @@
...
+- Pull from Jira acceptance criteria when available — translate them into concrete reviewer steps
+- Focus on functional verification: navigation flows, data correctness, edge cases
+- Include API verification steps if relevant (e.g., "call GET /api/x and confirm the response includes field Y")
+- **Do NOT suggest running unit tests, e2e tests, or lint commands** — CI covers those
+- Aim for **3–6 actionable steps**
+
+### Screenshots
```
Comment (hallieswan) [inline .claude/skills/pr-description/SKILL.md]: Should we change this to `Preview` to align with the template? 

```suggestion
### Preview
```

**Review by hallieswan — APPROVED** [review]
Looks good to me! Just a few minor suggestions

**Review by rrchai — APPROVED** [review]

**Review by hallieswan — APPROVED** [review]
